app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt

```diff
@@ -105,79 +123,123 @@ fun ServicesScreen(
 @Composable
 fun TopSection(
     searchViewModel: SearchServicesViewModel,
+    seekerProfileViewModel: SeekerProfileViewModel,
     listProviderViewModel: ListProviderViewModel,
     navigationActions: NavigationActions
 ) {
   val searchText by searchViewModel.searchText.collectAsState()
   val searchResults by searchViewModel.servicesList.collectAsState()
   val isSearching by searchViewModel.isSearching.collectAsState()
+  val userProfile by seekerProfileViewModel.seekerProfile.collectAsState()
+  val address = "EPFL - École Polytechnique Fédérale de Lausanne"
 
-  Column(
+  Box(modifier = Modifier.fillMaxWidth().testTag("servicesScreenTopSection")) {
+    // Background Image
+    Image(
+        painter = painterResource(id = R.drawable.top_background),
+        contentDescription = null,
+        contentScale = ContentScale.Crop,
+        modifier = Modifier.matchParentSize())
+    Column(modifier = Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
+      val context = LocalContext.current
+      val toast = Toast.makeText(context, "Not implemented yet", Toast.LENGTH_SHORT)
+      Row(
+          modifier = Modifier.fillMaxWidth().padding(start = 16.dp, end = 16.dp).height(45.dp),
+          horizontalArrangement = Arrangement.SpaceBetween,
+      ) {
+        Image(
+            painterResource(id = R.drawable.ic_user),
+            contentDescription = "profile picture",
+            Modifier.size(40.dp)
+                .clip(CircleShape)
+                .clickable { navigationActions.navigateTo(Route.PROFILE) }
+                .testTag("servicesScreenProfileImage"))
+        Column(horizontalAlignment = Alignment.Start) {
+          // User's Location
+          Row(verticalAlignment = Alignment.CenterVertically) {
+            Icon(
+                imageVector = Icons.Outlined.LocationOn,
+                contentDescription = "Location Icon",
+                tint = OnPrimary,
+                modifier =
+                    Modifier.size(16.dp).padding(end = 4.dp).testTag("servicesScreenLocationIcon"))
+
+            Text(
+                text =
+                    if (address.length > 27) { // Replace with `userProfile.address`
+                      address.take(27) + "..."
```

Comment:
did you think about using the general current Location ?

Reply:
It doesn't work unfortunately, its a backend problem